### PR TITLE
MM-19588 WCAG issues related to the Channel Header and its subsections

### PR DESCRIPTION
**Pull request by sudheerDev**
#### Summary
  * Fixes various WCAG issues in channel header section
  * Add aria-label for edit channel purpose modal
  * Add aria-label for rename channel modal
  * Add tab index to be negative if an element is hidden by aria
  * Add labeled by for search input
 + more based on feedback from AXE

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-19588

### CODE REVIEW

File: components/setting_item_max.jsx [hunk 1]
```diff
@@ -267,7 +267,7 @@ export default class SettingItemMax extends React.PureComponent {
                     <ul
                         tabIndex='-1'
                         ref={this.settingList}
-                        role='presentation'
+                        role='list'
```
Comment (devinbinnie) [inline components/setting_item_max.jsx]: @asaadmahmood Does this make sense to you?

File: components/search_bar/search_bar.jsx [hunk 2]
```diff
@@ -252,11 +254,11 @@ export default class SearchBar extends React.Component {
                     className='search-form__container'
                 >
                     <form
-                        role='application'
```
Comment (devinbinnie) [inline components/search_bar/search_bar.jsx]: @asaadmahmood Are there any ramifications to removing this?

File: components/setting_item_max.jsx (hunk 1 above)
Comment (asaadmahmood) [reply components/setting_item_max.jsx]: @devinbinnie Nops, I think we should avoid this change now not to accidentally change the readouts, we don't even need <ul>'s in there, and we can probably create a ticket to remove it. So we can avoid this for now.

File: components/search_bar/search_bar.jsx (hunk 2 above)
Comment (asaadmahmood) [reply components/search_bar/search_bar.jsx]: I think one of our major customers was having an issue with the cursor mode changing from section to section, so I would say we keep it.
Plus, there is an autocomplete there, which would stop working if we make this change (in terms of readout etc).

**Comment by sudheerDev** [comment]
@asaadmahmood PTAL.

**Pull request merged**